src/sas/qtgui/Plotting/Plotter.py

```diff
@@ -663,8 +670,12 @@ def onFitDisplay(self, fit_data):
         self.fit_result.title = 'Fit'
         self.fit_result.name = 'Fit'
 
-        # Plot the line
-        self.plot(data=self.fit_result, marker='-', hide_error=True)
+        if self.fit_result.name in self.plot_dict.keys():
```

Comment:
This is fine but it just made me think if the data explorer name change fix also updates the plot dictionary.
If not, then datasets with manually changed names won't get replotted properly with retained dimension.

However, this is related to the name change functionality, not this issue so let's merge this commit and potentially address the plot_dict update later as a separate issue.